### PR TITLE
Add AWS authentication with OIDC

### PR DESCRIPTION
**Pull request by reakaleek**
Tested in https://github.com/elastic/observability-github-secrets/commit/4b34ebc2574038d863a35f454a4d362605fb6bc9

### CODE REVIEW

File: hooks/pre-command [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+aws_account_id="697149045717"
```
Comment (v1v) [inline hooks/pre-command]: Do you think this ID can be customised?

File: hooks/pre-command [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+hash=$(echo -n "$(echo "${BUILDKITE_REPO}" | awk -F'[:.]' '{ printf $3 }')/${BUILDKITE_PIPELINE_SLUG}" | sha256sum | cut -c1-55)
```
Comment (v1v) [inline hooks/pre-command:9]: Can you add a comment about the rationale?

File: hooks/pre-command [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+aws_role_arn="arn:aws:iam::${aws_account_id}:role/bk-${hash}-role"
+duration_seconds=${BUILDKITE_PLUGIN_OBLT_AWS_AUTH_DURATION:-3600}
```
Comment (v1v) [inline hooks/pre-command:11]: Can you add a README so it does say about the default values?

File: hooks/pre-command [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+aws_account_id="697149045717"
+hash=$(echo -n "$(echo "${BUILDKITE_REPO}" | awk -F'[:.]' '{ printf $3 }')/${BUILDKITE_PIPELINE_SLUG}" | sha256sum | cut -c1-55)
+aws_role_arn="arn:aws:iam::${aws_account_id}:role/bk-${hash}-role"
```
Comment (v1v) [inline hooks/pre-command:10]: Is this our standard ARN format? If so, maybe it might be relevant to explain this?

File: hooks/pre-command [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+duration_seconds=${BUILDKITE_PLUGIN_OBLT_AWS_AUTH_DURATION:-3600}
+
+echo "~~~ :buildkite: Requesting OIDC token from Buildkite"
+TMPDIR=$(mktemp -d 2>/dev/null || mktemp -d -t 'oblt-aws-auth-buildkite-plugin')
+buildkite-agent oidc request-token --audience sts.amazonaws.com > "$TMPDIR"/token.txt
```
Comment (v1v) [inline hooks/pre-command]: I don't know if the token is something valid, but do you think we can use an in-mem variable so we don't need to delete the file after using it? 

```suggestion
TOKEN=$(buildkite-agent oidc request-token --audience sts.amazonaws.com)
```

**Review by v1v — COMMENTED** [review]
LGTM. I miss a bit of the context about our opinionated approach

File: hooks/pre-command (hunk 5 above)
Comment (reakaleek) [reply hooks/pre-command]: https://github.com/elastic/oblt-aws-auth-buildkite-plugin/pull/1/commits/c3b0fef46f2302b270a98021f827985f189fb7d4

File: hooks/pre-command (hunk 3 above)
Comment (reakaleek) [reply hooks/pre-command:11]: https://github.com/elastic/oblt-aws-auth-buildkite-plugin/pull/1/commits/18a8a7773809e4985cc095096a817b33a3f4bd31

File: hooks/pre-command (hunk 1 above)
Comment (reakaleek) [reply hooks/pre-command]: https://github.com/elastic/oblt-aws-auth-buildkite-plugin/pull/1/commits/30574d2f09e90491e8688c9fe6ca983189938954

File: hooks/pre-command (hunk 4 above)
Comment (reakaleek) [reply hooks/pre-command:10]: https://github.com/elastic/oblt-aws-auth-buildkite-plugin/pull/1/commits/341685cd11edfff482a611144f3354951422610e

File: hooks/pre-command (hunk 2 above)
Comment (reakaleek) [reply hooks/pre-command:9]: https://github.com/elastic/oblt-aws-auth-buildkite-plugin/pull/1/commits/341685cd11edfff482a611144f3354951422610e

**Review by v1v — APPROVED** [review]

**Pull request merged**